### ISSUE TITLE
Promise rejection: build Error: EMFILE: too many open files

### ISSUE THREAD

**Issue by nehaljwani** [microsoft/vscode-cmake-tools#2288]
On a project with a _huge_ number of build targets and with a capped `ulimit` on `nofile`, the CMake configure stage is marked as failed (although CMake itself succeeds). And on triggering the build step, it fails with:

```
[rollbar] Unhandled exception: Unhandled Promise rejection:
build Error: EMFILE: too many open files,
open '/path/to/builddir/.cmake/api/v1/reply/target-<redacted>o-Release<hash>.json' {}
```

The problem is that the following snippet:
https://github.com/microsoft/vscode-cmake-tools/blob/d2d98f80067738ae338c6e70dba7689012cd2e42/src/drivers/cmakefileapi/api_helpers.ts#L159-L161

Calls the following for each file with no consideration to limits and no retries:
https://github.com/microsoft/vscode-cmake-tools/blob/d2d98f80067738ae338c6e70dba7689012cd2e42/src/drivers/cmakefileapi/api_helpers.ts#L128

Yes, this problem can be worked around with increasing the `sysctl` for `fs.inotify.max_user_watches` or updating `/etc/security/limits.conf`, but that assumes that every developer has admin access or can get an admin to make this change for them, which is not always possible.

I found several discussions at https://stackoverflow.com/q/8965606/1005215 , perhaps this project can adopt one of them?

**Comment by bobbrow** [comment]
Thanks for reporting this.  We'll see what we can do, but will also accept a PR if someone who has a repro wants to help out.

**Comment by bobbrow** [comment]
This was fixed yesterday and is available in today's pre-release build, version 1.11.2.
